src/libcore/ptr/mod.rs

```diff
@@ -1074,17 +1074,23 @@ impl<T: ?Sized> *const T {
     /// operation because the returned value could be pointing to invalid
     /// memory.
     ///
-    /// When calling this method, you have to ensure that if the pointer is
-    /// non-NULL, then it is properly aligned, dereferencable (for the whole
-    /// size of `T`) and points to an initialized instance of `T`. This applies
-    /// even if the result of this method is unused!
+    /// When calling this method, you have to ensure that *either* the pointer is NULL *or* 
+    /// all of the following is true:
+    /// - it is properly aligned
+    /// - it must point to an initialized instance of T; in particular, the pointer must be 
+    ///   "dereferencable" in the sense defined at https://doc.rust-lang.org/nightly/core/ptr/index.html#safety
```

Comment:
Please don't use bare URL links, but show some nice text instead. Also please don't use absolute links but relative links (when we are e.g. on stable, we'd want to link to the stable docs, not the nightly ones.)